app.js

```diff
@@ -0,0 +1,92 @@
+const billAmount = document.querySelector(".bill-amount");
+const tips = document.querySelectorAll(".tip");
+const customTip = document.querySelector(".custom");
+const numberOfPeople = document.querySelector(".people-tip");
+const tipAmount = document.querySelector(".total-tip");
+const totalTip = document.querySelector(".total");
+const reset = document.querySelector("button");
+
+let totalBill = 0;
+let numberOfPersons = 1;
+let tip = 0;
+let tipsArray = Array.from(tips);
+
+const billValueHandler = (e) => {
+  totalBill = Number(e.target.value);
+  calculateTipAmount();
+};
+
+const numberOfPeopleHandler = (e) => {
+  numberOfPersons = Number(e.target.value);
+  calculateTipAmount();
+};
+
+const tipsHandler = (e) => {
+  tipsArray.forEach((element) => {
+    element.addEventListener("click", () => {
+      tipsArray.forEach((item) => {
+        if (item !== element) {
+          item.classList.remove("active");
+          customTip.value = "";
+        }
+      });
+      element.classList.add("active");
+
+      switch (element.textContent) {
+        case "5%":
+          tip = Number(5 / 100);
+          break;
+        case "10%":
+          tip = Number(10 / 100);
+          break;
+        case "15%":
+          tip = Number(15 / 100);
+          break;
+        case "25%":
+          tip = Number(25 / 100);
+          break;
+        case "50%":
+          tip = Number(50 / 100);
+          break;
+      }
+      calculateTipAmount();
+    });
+  });
+};
+
+const calculateTipAmount = (e) => {
```

Comment:
I think this function could be either broken into multiple functions that do things individually (one that calculates the tip amount, the other that calculates the total tip value and another one that updates the ui) or update its name to be more clear that is updates the ui, maybe something like updateTipUiData or something like that.

Reply:
Done!